pkg/plugin/GlobalPluginService.go

```diff
@@ -1948,3 +1930,330 @@ func (impl *GlobalPluginServiceImpl) MigratePluginDataToParentPluginMetadata(plu
 	}
 	return nil
 }
+
+func (impl *GlobalPluginServiceImpl) GetAllPluginMinData() ([]*bean2.PluginMinDto, error) {
+	pluginsParentMinData, err := impl.globalPluginRepository.GetAllPluginMinData()
+	if err != nil {
+		impl.logger.Errorw("GetAllPluginMinData, error in getting all plugin parent metadata min data", "err", err)
+		return nil, err
+	}
+	pluginMinList := make([]*bean2.PluginMinDto, 0, len(pluginsParentMinData))
+	for _, item := range pluginsParentMinData {
+		if item.Type == repository.PLUGIN_TYPE_PRESET {
+			continue
+		}
```

Comment:
add comment on why we are getting only non preset plugins. Also If it's for serving some specific purpose then change the method name and routes also